### PR TITLE
First Commit README.md on v2.0

### PR DESCRIPTION
**Pull request by gatonero**
Hopefully now on v2.0!

### CODE REVIEW

**Comment by jeroenterheerdt** [comment]
yes, this is on the right branch! awesome job!

File: README.md [hunk 1]
```diff
@@ -14,243 +14,222 @@
...
+
+This component calculates the time to run your irrigation system to compensate for moisture lost by evaporation / evapotranspiration. Using this component you water your garden, lawn or crops precisely enough to compensate what has evaporated. It takes into account precipitation (rain,snow) and adjusts accordingly, so if it rains or snows less or no irrigation is required. Multiple zones can be supported as each zone will have its own flow configuration. 
 
 > **Note - use this component at your own risk - we do not assume responsibility for any inconvience caused by using this component. Always use common sense before deciding to irrigate using the calculations this component provides. For example, irrigating during excessive rainfall might cause flooding. Again - we assume no responsibility for any inconvience caused.**
 
-> **Note If you want to go back and change your settings afterwards, you can either delete the instance and re-create it or edit the entity registry under config/.storage at your own risk.**
+> **Note If you want to go back and change your settings afterwards, you can either delete the zone and re-create it.**
```
Comment (jeroenterheerdt) [inline README.md:23]: looking at this again, what do we mean here? You can just edit a zone now. Not sure this is relevant. maybe remove?

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]
just one small comment

File: README.md (hunk 1 above)
Comment (gatonero) [reply README.md:23]: removed

**Comment by jeroenterheerdt** [comment]
awesome, let me know when you are ready to have this merged.

**Comment by gatonero** [comment]
Search for ???. Isn't clearfor me.

File: README.md [hunk 2]
```diff
@@ -14,243 +14,220 @@
...
+- Automatic weather data update: If enabled specify how often sensor update should happen (minutes, hours, days). Warning: weatherdata update time on or after calculation time! 
+
+#### ZONES
+Specify one or more irrigation zones here. The component calculates irrigation duration per zone, depending on size, throughput, state, module and mapping. A zone can be _disabled_ (so it doesn't do anything), _automatic_ or _manual_. If in automatic, duration is automatically calculated. If in manual, you specify the duration yourself. If disabled, the zone is not included in any calculations.
+
+> **When entering any values in the configuration of this component, keep in mind that the component will expect inches, sq ft, gallons, gallons per minute, or mm, m<sup>2</sup>, liters, liters per minute respectively depending on the settings in Home Assistant (imperial vs metric system).
+
+- Name: The name of your zone, e.g. garden
+- Size: The size of this zone (m<sup>2</sup> or sq ft)
+- Throughput: The flow of this zone (liter/minute or ???/minute)
```
Comment (jeroenterheerdt) [inline README.md:103]: gallon per minute

File: README.md [hunk 3]
```diff
@@ -14,243 +14,222 @@
...
+#### MODULES
+Add one or more modules that calculate irrigation duration. Each module comes with its own configuration and can be used to calculate duration for one or more zones. Modules can't be deleted if they are used by one or more zones.
+
+- 0:PyETO: Calculate duration based on the FAO56 calculation from the PyETO library. 
+  - If you set PyETO to not estimate, it will look for a solar radiation sensor in the mapping and will use that value. If there is none, it will use OWMs value (assuming you have OWM configured).
+If you let PyETO to estimate from temperature or sun hours, it will not ask OWM for a solar radiation value nor will it ask a sensor for a solar radiation value, even if you configured it in the mapping.
+  - Coastal: If the location you are tracking is situated on or adjacent to coast of a large land mass or anywhere else where air masses are influenced by a nearby water body, enable this setting.
+  - Solrad behaviour: Should solar radiation estimated from temperature or sun hours or disabled 
+  - Forecast days: How many forecast days taken into account
+- 1:Static: 'Dummy' module with a static configurable duration.
```
Comment (jeroenterheerdt) [inline README.md:130]: this should be static configurable netto precipitation

File: README.md [hunk 4]
```diff
@@ -14,243 +14,222 @@
...
+  - Delta: ???
```
Comment (jeroenterheerdt) [inline README.md:131]: delta = netto precipitation.

File: README.md [hunk 5]
```diff
@@ -14,243 +14,222 @@
...
+- State: Wether Updating and Calculation of that zone should be enabled in mode 'Automatic' or 'Manual' or 'Disabled'
+- Module: Choose the calculation module (see below) shall be used for that zone to calculate irrigation duration
+- Mapping: Which sensor mapping (see below) shall be used for that zone
+- Bucket: Either calculated or manualy set. Bucket >=0 no irrigation is necesarry, bucket <0 irrigation is necesarry.
+- Multiplier: Multiplies the duration of irrigation
+- Duration: Either calculated or manualy set
+
+Below each zone there are some buttons to update, calculate or delete that zone.
+
+#### MODULES
```
Comment (jeroenterheerdt) [inline README.md:121]: also add the passthrough module!

File: README.md [hunk 6]
```diff
@@ -14,243 +14,222 @@
...
+  - Coastal: If the location you are tracking is situated on or adjacent to coast of a large land mass or anywhere else where air masses are influenced by a nearby water body, enable this setting.
+  - Solrad behaviour: Should solar radiation estimated from temperature or sun hours or disabled 
+  - Forecast days: How many forecast days taken into account
+- 1:Static: 'Dummy' module with a static configurable duration.
+  - Delta: ???
+
+#### MAPPINGS
+For sensor configuration take care to make sure the unit the component expects is the same as your sensor provides.**
+
 - Names of sensors that supply required measurements (optional). Only required in mode 2) and 3). See [Measurements and Units](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Measurements-and-Units) for more information on the measurements and units expected by this component.
```
Comment (jeroenterheerdt) [inline README.md:136]: I think we should not talk about modes any more. Also, wiki needs updating as the units are now configurable on the mapping, as well as the aggregation.

File: README.md [hunk 7]
```diff
@@ -14,243 +14,222 @@
...
  
-**Multi-zone support**: For irrigation systems that have multiple zones which you want to run in series (one after the other), you need to add an instance of this integration *for each zone*. Of course, the configuration should be done for each zone, including the area the zone covers and the sprinkler settings.
+**Multi-zone support**: For irrigation systems that have multiple zones which you want to run in series (one after the other), you need to add a zone *for each zone*. Of course, the configuration should be done for each zone, including the area the zone covers and the corresponding settings.
 
- > **When entering any values in the configuration of this component, keep in mind that the component will expect inches, sq ft, gallons, gallons per minute, or mm, m<sup>2</sup>, liters, liters per minute respectively depending on the settings in Home Assistant (imperial vs metric system).
-For sensor configuration take care to make sure the unit the component expects is the same as your sensor provides.**
+ > **If you want to go back and change your settings afterwards, you can either update the zon or delete the zone and re-create it.**
```
Comment (jeroenterheerdt) [inline README.md:140]: the zon**e**

File: README.md [hunk 8]
```diff
@@ -14,243 +14,222 @@
...
-|`smart_irrigation.[instance]_calculate_daily_adjusted_run_time`|Triggers the calculation of daily adjusted run time. Use only if you disabled automatic refresh in the options.|
-|`smart_irrigation.[instance]_calculate_hourly_adjusted_run_time`|Triggers the calculation of hourly adjusted run time. Use for debugging only.|
-|`smart_irrigation.[instance]_enable_force_mode`|Enables force mode.|
-|`smart_irrigation.[instance]_disable_force_mode`|Disables force mode.|
-|`smart_irrigation.]instance]_reset_bucket`|Resets the bucket to 0. Needs to be called after done irrigating (see below).|
-|`smart_irrigation.[instance]_set_bucket`|Sets the bucket to the provided `value`. Use for debugging only.|
-
-#### Events
-The component uses a number of events internally that you do not need to pay attention to unless you need to debug things. The exception is the `_start` event.
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all zones. Use only if you disabled automatic refresh in the options.|
```
Comment (jeroenterheerdt) [inline README.md:154]: all automatic zones

File: README.md [hunk 9]
```diff
@@ -14,243 +14,222 @@
...
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: reset_all_buckets`|Resets all buckets.|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: set_all_buckets`|Sets all buckets to a specific `value`.|
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: update_all_zones`|Updates all zones with weather data|
```
Comment (jeroenterheerdt) [inline README.md:160]: all automatic zones

File: README.md [hunk 10]
```diff
@@ -14,243 +14,222 @@
...
-|`[instance]_bucketUpd`|Fired when the bucket is calculated. Happens at automatic refresh time or as a result of the `reset_bucket`, `set_bucket` or `calculate_daily_adjusted_run_time` service.|
-|`[instance]_forceModeTog`|Fired when the force mode is disabled or enabled. Result of calling `enable_force_mode` or `disable_force_mode`|
-|`[instance]_hourlyUpd`|Fired when the hourly adjusted run time is calculated. Happens approximately every hour and when `calculate_hourly_adjusted_run_time` service is called.|
+|`[instance]_start`|Fires depending on `sensor.[zone_name]` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
 
 #### Entities
-#### `sensor.smart_irrigation_base_schedule_index`
+#### `sensor.[zone_name]`
 The number of seconds the irrigation system needs to run assuming maximum evapotranspiration and no rain / snow. This value and the attributes are static for your configuration.
```
Comment (jeroenterheerdt) [inline README.md:170]: delete - this is about BSI which is not there any more.

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]
some more comments

File: README.md (hunk 2 above)
Comment (gatonero) [reply README.md:103]: Done

File: README.md (hunk 3 above)
Comment (gatonero) [reply README.md:130]: done

File: README.md (hunk 4 above)
Comment (gatonero) [reply README.md:131]: done

File: README.md (hunk 5 above)
Comment (gatonero) [reply README.md:121]: Never used., but found it
Passthrough module that returns the value of an Evapotranspiration sensor as delta.

File: README.md (hunk 6 above)
Comment (gatonero) [reply README.md:136]: Removed it. Will have a look at wiki.

File: README.md (hunk 7 above)
Comment (gatonero) [reply README.md:140]: :-)

File: README.md (hunk 8 above)
Comment (gatonero) [reply README.md:154]: done

File: README.md (hunk 9 above)
Comment (gatonero) [reply README.md:160]: done

File: README.md (hunk 10 above)
Comment (gatonero) [reply README.md:170]: done
With v2.0 README is getting shorter and shorter. Good job! :-)

File: README.md (hunk 9 above)
Comment (gatonero) [reply README.md:160]: and, all automatic buckets

File: README.md (hunk 10 above)
Comment (jeroenterheerdt) [reply README.md]: well, our wiki will grow and grow :)

File: README.md [hunk 11]
```diff
@@ -104,17 +105,17 @@ Specify one or more irrigation zones here. The component calculates irrigation d
 - Bucket: You can manipulate this calculated value.
 - Lead Time: In seconds. Time needed to warm up your irrigation system, e.g. time to establish a connection or start a pump etc.
 
-You can Update all zones and Callculate all.
+You can update and callculate all automatic zones.
```
Comment (jeroenterheerdt) [inline README.md:108]: cal**l**ulate

File: README.md [hunk 12]
```diff
@@ -104,17 +105,17 @@ Specify one or more irrigation zones here. The component calculates irrigation d
...
 
 **Per Zone Settings**
 You can change any value mentioned before. Additionaly there are some more options.
 
-- State: Wether Updating and Calculation of that zone should be enabled in mode 'Automatic' or 'Manual' or 'Disabled'
-- Module: Choose the calculation module (see below) shall be used for that zone to calculate irrigation duration
-- Mapping: Which sensor mapping (see below) shall be used for that zone
+- State: Wether updating and calculation of that zone should be enabled in mode 'Automatic' or 'Manual' or 'Disabled'.
```
Comment (jeroenterheerdt) [inline README.md:113]: This is not what State is for.
If a zone is in automatic, it is indeed part of the auto update and auto calc. In other words, weather data is automatically collected and the duration is automatically calculated. In Manual mode, that doesn't happen and the user can enter a fixed duration (this is the replacement for the force mode in v1). In Disabled the zone doesn't have a duration and you can't set one either.

File: README.md [hunk 13]
```diff
@@ -104,17 +105,17 @@ Specify one or more irrigation zones here. The component calculates irrigation d
...
+- Module: Choose the calculation module (see below) shall be used for that zone to calculate irrigation duration.
+- Mapping: Which sensor mapping (see below) shall be used for that zone.
 - Bucket: Either calculated or manualy set. Bucket >=0 no irrigation is necesarry, bucket <0 irrigation is necesarry.
-- Multiplier: Multiplies the duration of irrigation
-- Duration: Either calculated or manualy set
+- Multiplier: Multiplies the duration of the irrigation.
```
Comment (jeroenterheerdt) [inline README.md:117]: ... or divides if you do 0.5 for example :)

File: README.md [hunk 14]
```diff
@@ -104,17 +105,17 @@ Specify one or more irrigation zones here. The component calculates irrigation d
...
+- Duration: Either calculated or manualy set.
```
Comment (jeroenterheerdt) [inline README.md:118]: manual**l**y

File: README.md [hunk 15]
```diff
@@ -132,9 +133,9 @@ If you let PyETO to estimate from temperature or sun hours, it will not ask OWM
 - Passthrough: Passthrough module returns the value of an evapotranspiration sensor as delta.  
 
 #### MAPPINGS
-For sensor configuration take care to make sure the unit the component expects is the same as your sensor provides.**
+For sensor configuration take care to make sure the unit the component expects is the same as your sensor provides.
 
-**Multi-zone support**: For irrigation systems that have multiple zones which you want to run in series (one after the other), you need to add a zone *for each zone*. Of course, the configuration should be done for each zone, including the area the zone covers and the corresponding settings.
+**Multi-zone support**: For irrigation systems that have multiple zones which you want to run in series (one after the other or indepedenly), one zone must be created in each case. Of course, the configuration should be done for each zone, including the area the zone covers and the corresponding settings.
```
Comment (jeroenterheerdt) [inline README.md:138]: indepedenly --> independently.

File: README.md [hunk 16]
```diff
@@ -145,25 +146,7 @@ Links to wiki, forum and issues.
...
-|`Smart Irrigation: reset_all_buckets`|Resets all automatic buckets.|
-|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
-|`Smart Irrigation: set_all_buckets`|Sets all automatic buckets to a specific `value`.|
-
-#### Start Event
-
-| Event | Description|
-| --- | --- |
-|`[instance]_start`|Fires depending on `sensor.[zone_name]` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
+For each instance of the component the following entities, services and events will be available:
```
Comment (jeroenterheerdt) [inline README.md:149]: there is only one instance, so this first part can be updated.

File: README.md [hunk 17]
```diff
@@ -185,14 +168,34 @@ Sample screenshot:
...
+| Service | Description|
+| --- | --- |
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all automatic zones. Use only if you disabled automatic refresh in the options.|
+|`Smart Irrigation: update_zone`|Updates one specific zone with weather data|
+|`Smart Irrigation: update_all_zones`|Updates all automatic zones with weather data|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: reset_all_buckets`|Resets all automatic buckets.|
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: set_all_buckets`|Sets all automatic buckets to a specific `value`.|
```
Comment (jeroenterheerdt) [inline README.md:182]: all buckets are set, not just the buckets on the automatic zones

File: README.md [hunk 18]
```diff
@@ -185,14 +168,34 @@ Sample screenshot:
...
+#### Services
+
+| Service | Description|
+| --- | --- |
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all automatic zones. Use only if you disabled automatic refresh in the options.|
+|`Smart Irrigation: update_zone`|Updates one specific zone with weather data|
+|`Smart Irrigation: update_all_zones`|Updates all automatic zones with weather data|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: reset_all_buckets`|Resets all automatic buckets.|
```
Comment (jeroenterheerdt) [inline README.md:180]: same, not just automatic buckets are reset.

File: README.md [hunk 19]
```diff
@@ -185,14 +168,34 @@ Sample screenshot:
...
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: set_all_buckets`|Sets all automatic buckets to a specific `value`.|
+
+#### Start Event
+
+| Event | Description|
+| --- | --- |
+|`[instance]_start`|Fires depending on `sensor.[zone_name]` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
```
Comment (jeroenterheerdt) [inline README.md:188]: no, this event fires on the total of the durations of all non-disabled (so automatic or manual) zones, not just one. the math to determine when to fire this event is: sunrise - sum(duration for all non-disabled zones).

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]
some more comments :)

File: README.md (hunk 12 above)
Comment (gatonero) [reply README.md:113]: done

File: README.md (hunk 13 above)
Comment (gatonero) [reply README.md:117]: I waited for your  formulation. :-D

File: README.md (hunk 14 above)
Comment (gatonero) [reply README.md:118]: done, my poor english. ;-)

File: README.md (hunk 15 above)
Comment (gatonero) [reply README.md:138]: done

File: README.md (hunk 16 above)
Comment (gatonero) [reply README.md:149]: instance, object, class,entity, integration ... sometime I will learn it. :-D

For the the component the following entities, services and events will be available:

Ok?

File: README.md (hunk 17 above)
Comment (gatonero) [reply README.md:182]: learning by translation :-)

File: README.md (hunk 18 above)
Comment (gatonero) [reply README.md:180]: done

File: README.md (hunk 19 above)
Comment (gatonero) [reply README.md:188]: aha... would be fine in my issue. ;-)
done

File: README.md (hunk 16 above)
Comment (jeroenterheerdt) [reply README.md]: How about: "Once the component is installed, the following entities, services and events will be available:"?

File: README.md (hunk 11 above)
Comment (gatonero) [reply README.md]: done
having some beers ;-)

File: README.md [hunk 20]
```diff
@@ -14,243 +14,228 @@
...
+You can update and calculate all automatic zones.
 
-Sample screenshot:
+**Per Zone Settings**
+You can change any value mentioned before. Additionaly there are some more options.
 
-![](images/bsi_entity.png?raw=true)
+- State: 
+  - 'Automatic': Automatic updating and calculation of that zone. Module and sensor mapping is mandatory.
+  - 'Manual': Only manual updating and calculation of that zone. Module and sensor mapping is mandatory.
```
Comment (jeroenterheerdt) [inline README.md:115]: no module and sensor mapping is required in manual mode.

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]

File: README.md [hunk 21]
```diff
@@ -14,243 +14,228 @@
...
+5. Irrigation should be run for `sensor.[zone_name]`, which is 0 if `bucket`  >=0. Afterwards, the `bucket` needs to be reset (using `reset_bucket`). It's up to the user of the component to build the automation for this final step. See [Example automation](#step-4-creating-automation)
 
 There are many more options available, see below. To understand how `precipitation`, `netto precipitation`, the `bucket` and irrigation interact, see [example behavior on the Wiki](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Example-behavior-in-a-week).
 
 ## Operation modes
 You can use this component in various modes:
+
 1. **Full Open Weather Map**. In this mode all data comes from the Open Weather Map service. You will need to create and provide an API key. See [Getting Open Weater Map API Key](#getting-open-weather-map-api-key) below for instructions.
+
 2. **Full Sensors**. Using sensors. In this mode all data comes from sensors such as a weather station. When specificying a sensor for precipitation, note that it needs to be a cumulative daily sensor. Open Weather Map is not used and you do not need an API key.
```
Comment (jeroenterheerdt) [inline README.md:56]: maybe add that if you go sensors exclusively, there is no forecasting either.

File: README.md (hunk 16 above)
Comment (gatonero) [reply README.md]: Nice

File: README.md [hunk 22]
```diff
@@ -14,243 +14,228 @@
...
-4. **Not calculating**. This mode allows you to skip the calculations of evapotranspiration all together and instead rely on a sensor to provide this value. In this mode, you will need to specify a precipitation sensor as well. Both of these need to be a cumulative daily sensor.
 
-When planning to set up mode 2) (Full Sensors), 3) (Mixed) or 4) (Not Calculating) see [Measurements and Units](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Measurements-and-Units) for more information on the measurements and units expected by this component.
+3. **Mixed**. A combination of mode 1 and mode 2. In this mode part of the data is supplied by sensors and part by Open Weather Map. In this mode you will need to create and provide an API key. See [Getting Open Weater Map API Key](#getting-open-weather-map-api-key) below for instructions. When specificying a sensor for precipitation, note that it needs to be a cumulative daily sensor.
```
Comment (jeroenterheerdt) [inline README.md:58]: remove last sentence about "sensor for precipitation needing to be cumulative".

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]

File: README.md (hunk 20 above)
Comment (gatonero) [reply README.md:115]: done

File: README.md (hunk 21 above)
Comment (gatonero) [reply README.md:56]: Is it clear now?

File: README.md (hunk 22 above)
Comment (gatonero) [reply README.md:58]: done

File: README.md (hunk 21 above)
Comment (jeroenterheerdt) [reply README.md]: almost. For full sensors, we should add that you can't do forecasting either.

File: README.md (hunk 22 above)
Comment (jeroenterheerdt) [inline README.md]: no need to specify it is a cumulative daily sensor here.

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]

File: README.md (hunk 22 above)
Comment (gatonero) [reply README.md]: removed 'When specificying a sensor ...'

File: README.md (hunk 6 above)
Comment (gatonero) [reply README.md]: We have to find a way editing wiki

File: README.md (hunk 21 above)
Comment (gatonero) [reply README.md]: done

File: README.md [hunk 23]
```diff
@@ -31,6 +28,8 @@ The component uses the [PyETo module to calculate the evapotranspiration value (
 
 This is all the component does, and this is on purpose to provide maximum flexibility. Users are expected to use the value of `sensor.[zone_name]` to interact with their irrigation system and afterwards call the `smart_irrigation.reset_bucket` service. [See the example automations below](#step-4-creating-automations).
 
+Smart Irrigation custom component for Home Assistant. Partly based on the excellent work at https://github.com/hhaim/hass/.
```
Comment (jeroenterheerdt) [inline README.md:31]: I think we can remove this. None of his work is still present in v2. the BSI, HART, DART, WB stuff... all of it is gone.

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]
almost there. I recommend removing the refering to the other repo as we don't depend on their work any more.

File: README.md (hunk 23 above)
Comment (gatonero) [reply README.md]: done

**Review by jeroenterheerdt — APPROVED** [review]

**Pull request merged**